File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
from collections import Counter

def solution(array):
    counts = Counter(array).most_common()
    
    if not counts:
        return -1 
    
    max_count = counts[0][1]
    
    if len(counts) > 1 and counts[1][1] == max_count:
        return -1
    else:
        return counts[0][0]

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    const counts = new Map();

     for (const num of array) {
         counts.set(num, (counts.get(num) || 0) + 1);
    }

    if (counts.size === 0) return -1;
    
    const maxCount = Math.max(...counts.values());
    let mode = [];

    for (const [key, value] of counts) {
        if (value === maxCount) {
            mode.push(key);
        }
    }
    
    return mode.length === 1 ? mode[0] : -1;
}